internal/gps/solve_basic_test.go

```diff
@@ -1550,6 +1553,72 @@ func (sm *depspecSourceManager) ignore() map[string]bool {
 	return sm.ig
 }
 
+// DeduceConstraint tries to puzzle out what kind of version is given in a string -
+// semver, a revision, or as a fallback, a plain tag
+func (sm *depspecSourceManager) DeduceConstraint(s string, pi ProjectIdentifier) (Constraint, error) {
```

Comment:
FYI, I've made more changes to this function in #715. So whoever merges first, wins! 😉